cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -256,24 +297,25 @@ struct dispatch_t<StableAddress,
 
   template <typename ActivePolicy, size_t... Is>
   CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE cudaError_t
-    invoke_algorithm(::cuda::std::index_sequence<Is...>, ::cuda::std::integral_constant<Algorithm, Algorithm::prefetch>)
+  invoke_prefetch_algorithm(::cuda::std::index_sequence<Is...>, ActivePolicy policy)
   {
-    using policy_t          = typename ActivePolicy::algo_policy;
-    constexpr int block_dim = policy_t::block_threads;
+    const int block_dim = policy.BlockThreads();
 
     auto determine_config = [&]() -> cuda_expected<prefetch_config> {
       int max_occupancy = 0;
-      const auto error  = CubDebug(MaxSmOccupancy(max_occupancy, CUB_DETAIL_TRANSFORM_KERNEL_PTR, block_dim, 0));
+      auto error =
+        CubDebug(launcher_factory.MaxSmOccupancy(max_occupancy, kernel_source.TransformKernel(), block_dim, 0));
       if (error != cudaSuccess)
       {
         return ::cuda::std::unexpected<cudaError_t /* nvcc 12.0 fails CTAD here */>(error);
       }
-      const auto sm_count = get_sm_count();
```

Comment:
Is the definition of `get_sm_count` still needed, if you replaced it here?

Reply:
And what I don't understand as well: Why do you need to customize getting the SM count here, but you don't need to customize getting the max_shared_memory further above?

Reply:
Because (as mentioned in the description), I don't want to modify the UBLKCP code path quite just yet. The goal is to get `transform` exposed via the `prefetch` code path only at first, and then extend it to UBLKCP later. The latter introduces more complications that I'd rather tackle a bit later.

Reply:
`get_sm_count` is only used by the prefetch implementation, so you should be able to remove it now.

> I don't want to modify the UBLKCP code path quite just yet

Fair enough

Reply:
I've removed `get_sm_count`